tests/ada_grad_test.cpp

```diff
@@ -33,4 +33,4 @@ TEST_CASE("AdaGradLogisticRegressionTestFMat", "[AdaGradTest]")
 {
   AdaGrad adagrad(0.99, 32, 1e-8, 5000000, 1e-9, true);
   LogisticRegressionFunctionTest<arma::fmat>(adagrad, 0.003, 0.006);
-}
+}
```

Comment:
Maybe we should revert this?  I don't see a reason to remove the trailing newline :)

Reply:
Maybe I missed something but I think we don't use an extra trailing newline for other test files.

Reply:
Nevermind, I see what you mean, reverted with the last commit.